File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/level/block/entities/shrines/altar/CelestialAltarBlockEntity.java
package generations.gg.generations.core.generationscore.common.world.level.block.entities.shrines.altar;

import generations.gg.generations.core.generationscore.common.world.level.block.entities.GenerationsBlockEntities;
import generations.gg.generations.core.generationscore.common.world.level.block.entities.shrines.ShrineBlockEntity;
import generations.gg.generations.core.generationscore.common.world.level.block.shrines.CelestialAltarBlock;
import net.minecraft.core.BlockPos;
import net.minecraft.world.level.block.state.BlockState;

public class CelestialAltarBlockEntity extends ShrineBlockEntity {
    public CelestialAltarBlockEntity(BlockPos pos, BlockState state) {
        super(GenerationsBlockEntities.CELESTIAL_ALTAR.get(), pos, state);
    }

    @Override
    public String getVariant() {
        return getBlockState().getValue(CelestialAltarBlock.IS_SUN) ? "sun" : "moon";
    }
}

File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/level/block/entities/shrines/altar/CelestialAltarBlockEntity.kt
package generations.gg.generations.core.generationscore.common.world.level.block.entities.shrines.altar

import generations.gg.generations.core.generationscore.common.world.level.block.entities.GenerationsBlockEntities
import generations.gg.generations.core.generationscore.common.world.level.block.entities.shrines.ShrineBlockEntity
import generations.gg.generations.core.generationscore.common.world.level.block.shrines.CelestialAltarBlock
import net.minecraft.core.BlockPos
import net.minecraft.world.level.block.state.BlockState

class CelestialAltarBlockEntity(pos: BlockPos, state: BlockState) : ShrineBlockEntity(GenerationsBlockEntities.CELESTIAL_ALTAR, pos, state) {
    override fun getVariant(): String = if (blockState.getValue(CelestialAltarBlock.IS_SUN)) "sun" else "moon"
}
